web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.utils.ts

```diff
@@ -1,10 +1,48 @@
-import { BuyCreditsValues } from './BuyCreditsModal';
-
-export const getSellOrderLabel = (
-  sellOrder: BuyCreditsValues | undefined,
-): string => {
-  const { sellOrderId, price, askDenom = '', creditCount } = sellOrder || {};
-  return `${sellOrderId} (${price} ${askDenom
-    .substring(1)
-    .toUpperCase()}/credit: ${creditCount} credit(s) available)`;
+import { Option } from 'web-components/lib/components/inputs/SelectTextField';
+
+import { formatDenomText, microToDenom } from 'lib/denom.utils';
+
+import { ISellOrderInfo } from 'pages/Marketplace/Projects/Projects.types';
+
+import { BuyCreditsProject, BuyCreditsValues } from '..';
+
+export const getSellOrderLabel = (sellOrder: ISellOrderInfo): string => {
+  const { id, askAmount, askDenom = '', quantity } = { ...sellOrder };
+  const denom = formatDenomText(askDenom);
+  const price = microToDenom(askAmount);
+  return `${id} (${price} ${denom}/credit: ${quantity} credit(s) available)`;
+};
+
+export const getOptions = (project: BuyCreditsProject): Option[] => {
+  if (project?.sellOrders?.length) {
+    const sellOrderOptions = project?.sellOrders.map(sellOrder => {
+      return {
+        label: getSellOrderLabel(sellOrder),
+        value: sellOrder.id,
+      };
+    });
+    return [{ label: 'Choose a sell order', value: '' }, ...sellOrderOptions];
+  }
+
+  return [];
+};
+
+export const handleBuyCreditsSubmit = async (
+  values: BuyCreditsValues,
+  onSubmit?: (values: BuyCreditsValues) => Promise<void>,
+  selectedSellOrder?: ISellOrderInfo,
+): Promise<void> => {
+  if (!onSubmit || !selectedSellOrder) return Promise.reject();
```

Comment:
Might be useful to add an error message here. Also can throw an Error instead of returning `Promise.reject` (just preference - both do the same): 
```suggestion
  if (!onSubmit || !selectedSellOrder) throw new Error(’Some useful error msg’);
```

Reply:
done